site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,171 @@
+# Hosting: Firebase Functions
+
+This tutorial will guide you on how to deploy your bot to [firebase](https://firebase.google.com/) by using [Firebase Serverless Functions](https://firebase.google.com/docs/functions).
+
+## Prerequisites
+
+1. Google account (you can [create one here](https://accounts.google.com/signup)).
+
+## Setup
+
+This references the [official firebase documentation](https://firebase.google.com/docs/functions/get-started),
+so if you want a more detailed explanation of every step, you can take a look there.
+
+### 1. Create A Firebase Project
+
+1. Go to the [firebase console](https://console.firebase.google.com/) and click **Add Project**
+2. If prompted, review and accept the Firebase terms.
+3. Click Continue.
+4. Uncheck the analytics
+5. Click **Create Project**
+
+### 2. Setup Node.js and Firebase CLI
+
+You'll need a Node.js environment to write functions, and you'll need the Firebase CLI to deploy functions to the Cloud Functions runtime.
+>Important: Node.js versions 14, 16 and 18 are supported. See [Set runtime options](https://firebase.google.com/docs/functions/manage-functions#set_nodejs_version)
+for important information regarding ongoing support for these versions of Node.js.
+
+Once you have Node.js and npm installed, install the Firebase CLI via your preferred method. To install the CLI via npm, use:
+
+```shell
+npm install -g firebase-tools
+```
+
+### 3. Initialize The Project
+
+1. Run `firebase login` to log in via the browser and authenticate the Firebase CLI.
+2. Go to your Firebase project directory.
+3. Run `firebase init functions`, and type `y` when asked, whether you want to initialize a new codebase
+4. Choose `use existing project` and select the project you created in Step 1.
+5. The CLI gives you two options for language support:
+   - JavaScript
+   - TypeScript
+
+   For this tutorial, select TypeScript.
+
+6. Optionally you can select ESLint
+7. The CLI asks you if you want to install the dependencies with npm.
+   If you use another package manager like `yarn` or `pnpm` you can decline.
+   In that case you have to cd into the `functions` directory and install the dependencies manually.
+8. open `./functions/package.json` and look for the key: `"engines": {"node": "16"}`.
+   The node version should match your installed version of node.
+   Otherwise the project might not run.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point:
+
+```ts
+import * as functions from "firebase-functions";
+import {webhookCallback, Bot} from "grammy";
+
+const bot = new Bot(<BOT_TOKEN>);
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+//you can trigger your function on https://localhost/<firebase-projectname>/us-central1/helloworld during development 
+export const helloWorld = functions.https.onRequest(webhookCallback(bot));
+```
+
+## Development
```

Comment:
What if we move this development section after the deployment section?

Reply:
Not to judge from me to others, but i would only discover the development section after having deployed the function 500 times to test it :).
 The firebase docs also have this order and i think it makes sense, since you first develop and than deploy.

Reply:
I agree with @gertminov.